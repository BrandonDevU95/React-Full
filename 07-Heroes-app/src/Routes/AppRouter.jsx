import React from 'react';
import { Route, Routes, BrowserRouter } from 'react-router-dom';
import { DcScreen } from '../Components/DC/DcScreen';
import { LoginScreen } from '../Components/Login/LoginScree';
import { MarvelScreen } from '../Components/Marver/MarverlScreen';
import { SearchScreen } from '../Components/Search/SearchScreen';
import { Navbar } from '../Components/UI/Navbar';

export const AppRouter = () => {
   return (
      <BrowserRouter>
         <Navbar />
         <Routes>
            <Route path="/" element={<MarvelScreen />} />
            <Route path="/marvel" element={<MarvelScreen />} />
            <Route path="/dc" element={<DcScreen />} />
            <Route path="/search" element={<SearchScreen />} />
            <Route path="/login" element={<LoginScreen />} />
         </Routes>
      </BrowserRouter>
   );
};
