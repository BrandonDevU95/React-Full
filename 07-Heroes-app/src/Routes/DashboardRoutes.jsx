import { Route, Routes } from 'react-router-dom';
import { DcScreen } from '../Components/DC/DcScreen';
import { HeroScreen } from '../Components/Hero/HeroScreen';
import { MarvelScreen } from '../Components/Marver/MarverlScreen';
import { SearchScreen } from '../Components/Search/SearchScreen';
import { Navbar } from '../Components/UI/Navbar';

export const DashboardRoutes = () => {
   return (
      <>
         <Navbar />
         <div className="container">
            <Routes>
               <Route path="marvel" element={<MarvelScreen />} />
               <Route path="dc" element={<DcScreen />} />
               <Route path="search" element={<SearchScreen />} />
               <Route path="/hero" element={<HeroScreen />} />
               <Route path="/" element={<MarvelScreen />} />
            </Routes>
         </div>
      </>
   );
};
