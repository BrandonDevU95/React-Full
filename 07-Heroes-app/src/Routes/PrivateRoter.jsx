import React from "react";
import { useContext } from "react";
import { AuthContext } from "../Auth/authContext";
import { Navigate, useLocation } from "react-router-dom";

export const PrivateRoter = ({ children }) => {
   const { user } = useContext(AuthContext);
   const { pathname, search } = useLocation();

   localStorage.setItem("lastPath", pathname + search);

   return user.logged ? children : <Navigate to="/login" />;
};
