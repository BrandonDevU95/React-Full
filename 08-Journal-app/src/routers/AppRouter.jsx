import { AuthRouter } from "./AuthRouter";
import {
   BrowserRouter as Router,
   Switch,
   Route,
   Link,
   Redirect,
} from "react-router-dom";
import { JournalScreen } from "../components/journal/JournalScreen";

export const AppRouter = () => {
   return (
      <Router>
         <div>
            <Switch>
               <Route path="/auth" component={AuthRouter} />
               <Route path="/" exact component={JournalScreen} />
               <Redirect to="/auth/login" />
            </Switch>
         </div>
      </Router>
   );
};
