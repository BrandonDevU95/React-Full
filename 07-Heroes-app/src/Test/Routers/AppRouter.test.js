import { AppRouter } from "../../Routes/AppRouter";
import { AuthContext } from "../../Auth/authContext";
import { mount } from "enzyme";

describe("Pruebas en AppRouter", () => {
   const contextValue = {
      user: {
         logged: false,
      },
   };

   test("debe de mostrar login si no esta autenticado", () => {
      const wrapper = mount(
         <AuthContext.Provider value={contextValue}>
            <AppRouter />
         </AuthContext.Provider>
      );

      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find("h1").text().trim()).toBe("LoginScreen");
   });
});
