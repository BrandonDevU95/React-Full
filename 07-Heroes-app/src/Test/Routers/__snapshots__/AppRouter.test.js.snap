// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AppRouter debe de mostrar login si no esta autenticado 1`] = `
<div
  className="container mt-5 "
>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
